src/notification.c

```diff
@@ -47,130 +50,476 @@ struct _notification_private {
+        // some fields need escaped (format) and some raw (body, summary) data.
+        char * envValue = string_replace_all("\\\\", "\\", g_strdup(envValue_buff));
+
+
+        if( STR_EQ("DUNST_TIMEOUT", envName) ) {
+                gint64 v = g_ascii_strtoll(envValue, NULL, 10);
+                if (!v) {
+                        LOG_W("Script tried to set DUNST_TIMEOUT to invalid value (%s), ignoring", envValue);
+                } else {
+                        if (n->timeout/1000 != v) {
+                                LOG_I("Script overrides TIMEOUT: %ld -> %ld", n->timeout / 1000, v);
+                                n->timeout = v * 1000;
+                                ret = 1;
+                        }
+                }
+
+        } else if( STR_EQ("DUNST_PROGRESS", envName) ) {
+                int v = (int) g_ascii_strtoll(envValue, NULL, 10);
+                if (n->progress != v) {
+                        LOG_I("Script overrides PROGRESS: %d -> %d", n->progress, v);
+                        n->progress = v;
+                        ret = 1;
+                }
+
+        }  else if( STR_EQ("DUNST_TRANSIENT", envName) ) {
+                bool v = (g_ascii_strtoll(envValue, NULL, 10) > 0);
+                if (n->transient != v) {
+                        LOG_I("Script overrides TRANSIENT: %d -> %d", n->transient, v);
+                        n->transient = v;
+                        ret = 1;
+                }
+
+        } else if( STR_EQ("DUNST_SUMMARY", envName) ) {
+                c = UNESCAPE_NEWLINES(envValue);
+                OVERRIDE_IF_DIFFERENT_STR(n, summary, c);
+
+        } else if( STR_EQ("DUNST_BODY", envName) ) {
+                c = UNESCAPE_NEWLINES(envValue);
+                OVERRIDE_IF_DIFFERENT_STR(n, body, c);
+
+        } else if( STR_EQ("DUNST_FORMAT", envName) ) {
+        // Took me 3 goddamn hours to figure out that format is not allocated per
+        // notification - it is a pointer to some global value. And if you free
+        // it, you will screw up all the following notifications... FML!
+        if ( !STR_EQ(n->format, envValue) && (n->format && strlen(envValue)) ) {
+                LOG_I("Script overrides format: '%s' -> '%s'",n->format, envValue);
+                n->format = g_strndup(envValue, DUNST_NOTIF_MAX_CHARS);
+                ret = 1;
+        }
+
+        } else if( STR_EQ("DUNST_CATEGORY", envName) ) {
+                OVERRIDE_IF_DIFFERENT_STR(n, category, envValue);
+
+        } else if( STR_EQ("DUNST_STACK_TAG", envName) ) {
+                OVERRIDE_IF_DIFFERENT_STR(n, stack_tag, envValue);
+
+        } else if( STR_EQ("DUNST_ICON_PATH", envName) ) {
+                OVERRIDE_IF_DIFFERENT_STR(n, icon_path, envValue);
+
+        }  else if( STR_EQ("DUNST_URGENCY", envName) ) {
+                enum urgency u = URG_NONE;
+
+                // Dictionary would be nice, but hey...
+                if( STR_EQ("LOW", envValue) ) u = URG_LOW;
```

Comment:
There's a function for this: string_parse_enum (from option_parser.h). You can pass the urgency_enum_data (from settings_data.h) to it. The interface of the string_parse_enum function is a bit weird to handle errors (the return pointer isn't changed when an error occurs).

Reply:
Ah, nice. I was trying to look for it but it ended up as quite a rabbithole when I looked on the configuration/parameters :laughing:

Will do :+1: